EssentialsChat/src/main/java/com/earth2me/essentials/chat/processing/AbstractChatHandler.java

```diff
@@ -222,6 +235,27 @@ protected void handleChatRecipients(AsyncPlayerChatEvent event) {
         }
     }
 
+    /**
+     * Re-create type-based chat event from the base chat event, call it and mirror changes back to the base chat event.
+     * @param event Event based on which a type-based event will be created, and to which changes will be applied.
+     * @param chatType Chat type which determines which event will be created and called.
+     */
+    protected void callChatEvent(final AsyncPlayerChatEvent event, final ChatType chatType) {
```

Comment:
Compared to previously submitted code: added documentation and set privacy to `protected` to match other methods of `AbstractChatHandler`.